src/transforms.ts

```diff
@@ -141,6 +146,78 @@ function kebabToSnake<T>(obj: T): DeepKebabToSnake<T> {
   return deepTransformKeys(obj, toSnakeCase) as never
 }
 
+function transformFormData(
+  formData: FormData,
+  transformKey: (str: string) => string,
+) {
+  const transformed = new FormData()
+  for (const [key, value] of formData) {
+    transformed.append(transformKey(key), value)
+  }
+  return transformed
+}
+
+function transformURLSearchParams(
+  searchParams: URLSearchParams,
+  transformKey: (str: string) => string,
+) {
+  const transformed = new URLSearchParams()
+  for (const [key, value] of searchParams) {
+    transformed.append(transformKey(key), value)
+  }
+  return transformed
+}
+
+function transformQuery(
+  query: SearchParams,
+  transformKey: (str: string) => string,
+) {
+  if (Array.isArray(query)) {
+    return query.map(([key, value]) => [transformKey(key), value])
+  }
+
+  if (query instanceof URLSearchParams) {
+    return transformURLSearchParams(query, transformKey)
+  }
+
+  if (typeof query === 'string') {
+    const searchParams = new URLSearchParams(query)
+    return transformURLSearchParams(searchParams, transformKey).toString()
+  }
+
+  if (typeof query === 'object') return deepTransformKeys(query, transformKey)
+
+  return query
+}
+
+function transformBody(
+  body: EnhancedBodyInit | undefined,
+  transformKey: (str: string) => string,
+) {
+  if (body instanceof URLSearchParams) {
+    return transformURLSearchParams(body, transformKey)
+  }
+
+  if (body instanceof FormData) {
+    return transformFormData(body, transformKey)
+  }
+
+  if (typeof body === 'object') return deepTransformKeys(body, transformKey)
```

Comment:
Because `deepTransformKeys` already makes that check, we don't need to recheck it. I included new tests for those cases and simplified the code :) Thanks!